packages/server-wallet/src/wallet/__test__/fixtures/states.ts

```diff
@@ -34,28 +34,30 @@ const defaultState: State = {
 
 // Caching signatures saves about 200ms per signature
 // TODO: Persist these signatures between tests
-const signatureCache: Record<string, string> = {};
-const _signState = (s: State, pk: string): string => {
-  const key = `${pk}-${hashState(s)}`;
-  return (signatureCache[key] = signatureCache[key] || signState(s, pk));
+const signatureCache: Record<string, SignatureEntry> = {};
+const _signState = (s: State, sw: SigningWallet): SignatureEntry => {
+  const key = `${sw.privateKey}-${hashState(s)}`;
+  return (signatureCache[key] = signatureCache[key] || sw.signState(s));
 };
```

Comment:
If we're adding some caching, should we explore using `redis` instead of implementing our own strategy?

Reply:
Perhaps if this is just for local testing it doesn't matter.

Reply:
It was written at the time for local testing. I think we should even persist them between test runs -- it would mean a test file could run in 3s instead of 10s, which seems like a big win for developer experience